hal/common/SerialBase.cpp

```diff
@@ -26,10 +26,11 @@ SerialBase::SerialBase(PinName tx, PinName rx) :
                                                  _thunk_irq(this), _tx_usage(DMA_USAGE_NEVER),
                                                  _rx_usage(DMA_USAGE_NEVER),
 #endif
-                                                _serial(), _baud(9600) {
+                                                _serial(), _baud(MBED_CONF_CORE_STDIO_BAUD_RATE) {
     // No lock needed in the constructor
 
     serial_init(&_serial, tx, rx);
+    serial_baud(&_serial, _baud);
```

Comment:
@bogdanm originally said this:

> Before this PR, the default baud rate of a serial object was defined in its HAL implementation. The PR might break some existing code

This actually maybe problematic for Silicon Labs targets since their interface chips are tied to 115200 I believe. If we make this change, you might have to set the default stdio for Silicon Labs targets to be 115200 instead of 9600. @0xc0170 might know more about this.